model/job_test.go

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package model
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestJobAuditable(t *testing.T) {
+	job := &Job{
+		Id:             "arandomstring0123456789012",
+		Type:           JobTypeExportProcess,
+		Priority:       42,
+		CreateAt:       1336,
+		StartAt:        1337,
+		LastActivityAt: 1666609360813,
+		Status:         JobStatusInProgress,
+		Progress:       32,
+		Data:           StringMap{"Hello": "World"},
+	}
+
+	audit := job.Auditable()
+
+	require.Equal(t, job.Id, audit["id"])
+	require.Equal(t, job.Type, audit["type"])
+	require.Equal(t, job.Priority, audit["priority"])
+	require.Equal(t, job.CreateAt, audit["create_at"])
+	require.Equal(t, job.StartAt, audit["start_at"])
+	require.Equal(t, job.LastActivityAt, audit["last_activity_at"])
+	require.Equal(t, job.Status, audit["status"])
+	require.Equal(t, job.Progress, audit["progress"])
+	require.Equal(t, job.Data, audit["data"])
+}
+
+func TestJobIsValid(t *testing.T) {
+	t.Run("valid", func(t *testing.T) {
+		job := &Job{
+			Id:             "arandomstring0123456789012",
+			Type:           JobTypeExportProcess,
+			Priority:       42,
+			CreateAt:       1336,
+			StartAt:        1337,
+			LastActivityAt: 1666609360813,
+			Status:         JobStatusInProgress,
+			Progress:       32,
+			Data:           StringMap{"Hello": "World"},
+		}
+
+		require.Nil(t, job.IsValid())
```

Comment:
Nit: in all these tests, can we use `require.NoError` and `require.Error` instead of `require.Nil` and `require.NotNil`? It's the exact same check, but slightly more accurate semantically.

Reply:
I think `job.IsValid()` return `*model.AppError` hence it can't be checked with `require.Error`.

Reply:
Exactly that. The return value of `IsValid()` is a `*model.AppError`. Wrapped into an `error` this would be `(*model.AppError)(nil)`, instead of plain `nil`. It would always be interpreted as non-`nil`.

Reply:
Aaaaaah, sorry, my bad. Thanks for the clarification, guys :)